android-components/components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/manager/State.kt

```diff
@@ -52,10 +53,13 @@ import mozilla.components.service.fxa.FxaAuthData
  * State transitions are described by a transition matrix, which is described in [State.next].
  */
 
-internal enum class AccountState {
-    NotAuthenticated,
-    Authenticated,
-    AuthenticationProblem,
+private val logger = Logger("FirefoxAccountStateMachine")
+
+internal sealed class AccountState {
+    object NotAuthenticated : AccountState()
```

Comment:
Organization of sub-classes could be improved if we chose to follow alphabetic order or to prioritize it on the  frequency of occurrence.